packages/nouns-webapp/src/components/Auction/index.tsx

```diff
@@ -43,16 +43,22 @@ const Auction: React.FC<AuctionProps> = props => {
     currentAuction && history.push(`/noun/${currentAuction.nounId.toNumber() + 1}`);
   };
 
-  const handleKeyPress = useCallback(event => {
-    if (event.code === 'ArrowLeft') prevAuctionHandler();
-    else if (event.code === 'ArrowRight') nextAuctionHandler();
-    else return;
-  }, []);
+  const handleKeyPress = (event: KeyboardEvent) => {
+    if (currentAuction && lastNounId) {
+      const isFirstAuction = currentAuction.nounId.eq(0);
```

Comment:
`isFirstAuction` and `isLastAuction` are used in a few other places, part of the solution might be to memoize these and reuse the output